src/Polling/Common/TestService.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace TemporalioSamples.Polling.Common;
+
+public class TestService
+{
+    private int tryAttempt;
+    private int errorAttempts = 5;
+
+    public TestService()
+    {
+    }
+
+    public TestService(int errorAttempts) => this.errorAttempts = errorAttempts;
```

Comment:
```suggestion
    private readonly int errorAttempts;

    public TestService(int errorAttempts = 5) => this.errorAttempts = errorAttempts;
```

May be a tad clearer. Even if you want to keep the default constructor, make the field readonly